src/goTest.ts

```diff
@@ -191,7 +191,11 @@ export function goTest(testconfig: TestConfig): Thenable<boolean> {
 		outputChannel.appendLine('');
 
 		let proc = cp.spawn(goRuntimePath, args, { env: testEnvVars, cwd: testconfig.dir });
-		proc.stdout.on('data', chunk => outputChannel.append(chunk.toString()));
+		proc.stdout.on('data', chunk => {
+			let testOutput = expandFilePathInOutput(chunk.toString(), testconfig.dir);
```

Comment:
I think it requires a proposal to merge such a feature. Have you tried the `-exec` flag? It allows you to intercept the go test with any program you want. You might be able to access to the test file's absolute path with a custom executable.

A sample exec program: https://github.com/golang/go/blob/master/misc/android/go_android_exec.go